### PR TITLE
refactor: multiple code improvements

### PR DESCRIPTION
**Pull request by iFurySt**
* Ensure super().__init__() calls in class constructors
* Fix incorrect types
* Delete unused code
* Fix typos

### CODE REVIEW

File: opendevin/llm/llm.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ def attempt_on_error(retry_state):
                 f'{retry_state.outcome.exception()}. Attempt #{retry_state.attempt_number} | You can customize these settings in the configuration.',
                 exc_info=False,
             )
-            return True
+            return None
```
Comment (tobitege) [inline opendevin/llm/llm.py:180]: Shouldn't this stay True, otherwise the decorator would stop

File: opendevin/llm/llm.py [hunk 2]
```diff
@@ -233,7 +233,8 @@ def _post_completion(self, response: str) -> None:
         """
         try:
             cur_cost = self.completion_cost(response)
-        except Exception:
+        except Exception as e:
+            logger.warning(f'Cost calculation failed: {e}')
```
Comment (tobitege) [inline opendevin/llm/llm.py:237]: Please no logging on every completion, it should stay silent

File: opendevin/llm/llm.py [hunk 3]
```diff
@@ -303,9 +304,11 @@ def completion_cost(self, response):
                 )
                 self.metrics.add_cost(cost)
                 return cost
-            except Exception:
+            except Exception as e:
                 self.cost_metric_supported = False
-                logger.warning('Cost calculation not supported for this model.')
+                logger.warning(
```
Comment (tobitege) [inline opendevin/llm/llm.py:309]: Please no "e" here. This is a one-time logging, that is expected if litellm reports it back as unsupported.

File: opendevin/server/listen.py [hunk 4]
```diff
@@ -253,7 +253,6 @@ async def websocket_endpoint(websocket: WebSocket):
     """
     await websocket.accept()
 
-    session = None
```
Comment (tobitege) [inline opendevin/server/listen.py:256]: shouldn't this stay to reset it due to the potential return below?

**Review by tobitege — CHANGES_REQUESTED** [review]
In general good findings, but some parts I'd rather leave as they are.

**Comment by iFurySt** [comment]
okay, I'll update it tomorrow

File: opendevin/llm/llm.py (hunk 1 above)
Comment (iFurySt) [reply opendevin/llm/llm.py:180]: Actually, the `after` only accepts a function that returns None, as per its signature `after: (RetryCallState) -> Awaitable[None] | None`, the after callback can't control whether the retry stops 😅

you can give it a try if you like, i attach a demo below for you:
```python
import random
import logging
from tenacity import retry, retry_if_exception_type, stop_after_attempt, wait_random_exponential

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)


class CustomException(Exception):
    pass


def attempt_on_error(retry_state):
    logger.info(f'Retrying... Attempt #{retry_state.attempt_number}')
    return None

@retry(
    reraise=True,
    stop=stop_after_attempt(5),
    wait=wait_random_exponential(min=1, max=10),
    retry=retry_if_exception_type(CustomException),
    after=attempt_on_error
)
def unreliable_function():
    if random.random() < 0.7:
        logger.error("Function failed")
        raise CustomException("Network error")
    logger.info("Function succeeded")

try:
    unreliable_function()
except CustomException as e:
    logger.error(f"Final failure after retries: {e}")
```

File: opendevin/server/listen.py (hunk 4 above)
Comment (iFurySt) [reply opendevin/server/listen.py:256]: I believe it's pointless here.

File: opendevin/llm/llm.py (hunk 1 above)
Comment (tobitege) [reply opendevin/llm/llm.py:180]: Ok, thanks for the script. I wasn't sure about that syntax, thanks for clarifying it.

File: opendevin/server/listen.py (hunk 4 above)
Comment (tobitege) [reply opendevin/server/listen.py:256]: On second look, I did a search in the codebase and the method itself isn't referenced anywhere, unless I'm missing some automatic mechanism that picks it up? 🤔

**Comment by iFurySt** [comment]
@tobitege i've submitted the adjustments, please let me know if there are any issues.

**Review by tobitege — APPROVED** [review]
LGTM. Thanks! 👍

**Pull request merged**